### ISSUE TITLE
AlphaMode / alphaCutoff information is lost when re-exporting Project Polly in Blender 2.90

### ISSUE THREAD

**Issue by cx20** [KhronosGroup/glTF-Blender-IO#1264]
**Describe the bug**
I have tried loading the [Project Polly model](https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/master/polly/project_polly.blend) in Blender `2.90` and outputting it in glTF format.
However, some information seems to be missing. For example, the `alphaMode` / `alphaCutoff`.

**To Reproduce**
1. Download the following file.
https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/master/polly/project_polly.blend 
    (Probably created in Blender `2.79`)
2. run Blender `2.90`
3. Open `project_polly.blend`
4. Export to `project_polly.gltf`
   Format : [gltF Separate (.gltf + .bin + textures)]
   [x] Cameras
5. Check the contents of `project_polly.gltf`.
   `alphaMode` / `alphaCutoff` is not output.

|Material name|glTF JSON|
|:-------:|:-------:|
|Location Transparent|![image](https://user-images.githubusercontent.com/1833979/99792961-44298980-2b6b-11eb-96a9-e0aef4c2d98e.png)|
|Polly Display|![image](https://user-images.githubusercontent.com/1833979/99793004-54416900-2b6b-11eb-89b6-cfc2d4b31b72.png)|

**Expected behavior**

|Material name|glTF JSON|
|:-------:|:-------:|
|Location Transparent|![image](https://user-images.githubusercontent.com/1833979/99793198-9ec2e580-2b6b-11eb-917b-b5a6d6f68b0f.png)|
|Polly Display|![image](https://user-images.githubusercontent.com/1833979/99793269-c0bc6800-2b6b-11eb-806b-51a0f952b26b.png)|

The output results are expected to be similar to the glTF file below.
https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/master/polly/project_polly.gltf

**Screenshots**

|Blender 2.79 glTF Export results|Blender 2.90 glTF Export results|
|:------:|:------:|
|![image](https://user-images.githubusercontent.com/1833979/99793497-201a7800-2b6c-11eb-9215-609a87f35fa1.png)|![image](https://user-images.githubusercontent.com/1833979/99794579-b56a3c00-2b6d-11eb-89b3-00c564222c13.png)|

**.blend file/ .gltf**
https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/master/polly/project_polly.blend 

**Version**
 - OS: Windows 10 Version 1909
 - Blender Version 2.90.1

**Comment by scurest** [comment]
This uses the old "glTF Metallic Roughness" group. The docs for how alpha are supposed to be determined are [here](https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/dd7a3dbd8f43a79d572e7c45f4215f770bb92a37/docs/user.md#alpha).

**Comment by cx20** [comment]
@scurest Thank you for your advice.
I have seen that changing the default `Blend Mode` from `Opaque` to another value changes the `alphaMode` of glTF.
However, the value of `AlphaCutoff` `0.5` is not output to the glTF file and seems to be `0`. Is this the expected result?

|default Blend Mode|modified Blend Mode|glTF alphaMode|
|:---:|:---:|:---:|
|Opaque|Alpha Blend|BLEND|
|Opaque|Alpha Clip |MASK|

|Material name|Before|After|
|:-------:|:-------:|:-------:|
|Location Transparent|![image](https://user-images.githubusercontent.com/1833979/99802731-878bf400-2b7b-11eb-9a9b-d6fd2071a0af.png)|![image](https://user-images.githubusercontent.com/1833979/99802754-970b3d00-2b7b-11eb-86a0-0b5465b6bdca.png)|
|Polly Display|![image](https://user-images.githubusercontent.com/1833979/99802804-ac806700-2b7b-11eb-9921-f7a5f5dcd1b3.png)|![image](https://user-images.githubusercontent.com/1833979/99802819-b5713880-2b7b-11eb-87f3-5dda568dfd2e.png)|

|Material name|Before|After|
|:-------:|:-------:|:-------:|
|Location Transparent|![image](https://user-images.githubusercontent.com/1833979/99802937-e5b8d700-2b7b-11eb-8aa6-9b6483f4eb44.png)|![image](https://user-images.githubusercontent.com/1833979/99802951-ec474e80-2b7b-11eb-8276-4adb06c31f7d.png)|
|Polly Display|![image](https://user-images.githubusercontent.com/1833979/99802973-f6694d00-2b7b-11eb-90ee-fdceb2841bf0.png)|![image](https://user-images.githubusercontent.com/1833979/99802980-fc5f2e00-2b7b-11eb-9d7f-d6e4c1a6ba1a.png)|

**Comment by scurest** [comment]
I was just diagnosing the problem, sorry if I was unclear.

The exporter currently always uses "Blend Mode" to set alphaMode and (when using Alpha Clip) "Clip Threshold" to set alphaCutoff. Some parts of the old "glTF Metallic Roughness" group are implemented, like doubledSided, but the alpha stuff is not.

You can use "Clip Threshold" to set the alphaCutoff for the moment.

**Comment by cx20** [comment]
> You can use "Clip Threshold" to set the alphaCutoff for the moment.

Yes. I set the "Clip Threshold" and made sure it is reflected in the glTF "alphaCutoff".

|Blender|glTF|
|:----:|:----:|
|![image](https://user-images.githubusercontent.com/1833979/99806816-e48aa880-2b81-11eb-8b66-3c2bca184ce0.png)|![image](https://user-images.githubusercontent.com/1833979/99806845-ee141080-2b81-11eb-8a8c-98c17c899d44.png)|

**Comment by cx20** [comment]
I could see that there are compatibility issues with 2.79 and 2.90 so far.

BTW, I think it's another matter, but Polly exported in Blender 2.90 seems to pop its eyeballs.
![image](https://user-images.githubusercontent.com/1833979/99808122-bdcd7180-2b83-11eb-88a1-5fa7ecf3fae0.png)

**Comment by donmccurdy** [comment]
Sounds like we should update the source `.blend` sample file to be compatible with 2.90? At least for the material settings portion of the question.

**Comment by cx20** [comment]
@donmccurdy  Yes, I think it would be nice to have a `.blend` file for Blender 2.90.
[The Project Polly glTF fileglTF file](https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/master/polly/project_polly.gltf) in the old repository had a lot of validation errors, so I tried to re-export it with the latest version of Blender 2.90. But that didn't work either and I was having trouble with it.

**Comment by cx20** [comment]
BTW, the reason for this in the first place was because I wanted to know what was correct, as each viewer had a different result for the Project Polly nameplate display.

While the results of `Filament` are apparently the desired result, I feel that `Three.js` and `Babylon.js` are more correct in terms of glTF output. However, I have not been able to make a decision because I don't know the correct answer.

![image](https://user-images.githubusercontent.com/1833979/100494202-dab90480-3182-11eb-85e3-5b078fd2e1d0.png)

**Comment by emackey** [comment]
Not to make things more complicated, but Polly has always abused alpha blending to look like glass, which was never the intention of alpha in glTF.  Alpha is supposed to indicate the absence of a material, such as the edge of a leaf in a quad, or medical gauze wrap with lots of holes too small to see.

For glass, we recently ratified a new extension, [`KHR_materials_transmission`](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_transmission).  Among other things, this allows full specular reflections to take place on the clear parts of the glass (which alpha blending does not do).  It also allows for frosted glass (via material roughness), and lets path tracers know they can trace reflections in the glass.

Blender support for this is [documented in 2.91](https://docs.blender.org/manual/en/2.91/addons/import_export/scene_gltf2.html#transmission).

**Comment by cx20** [comment]
@emackey Thank you for telling me. 

I was wondering why the wooden part of the base of the nameplate was transparent, and then I realized it was because glass and wood are made of the same material. Perhaps it is better to separate this part of the material.
(Why the wood isn't transparent in the case of `Filament` is a curious but different topic).

And, I thought it would be nice to have a Polly v2 that takes advantage of the new glTF extension.
